src/Utils.js

```diff
@@ -10,6 +10,10 @@ import {
   REACT15,
 } from './version';
 
+function equality(a, b) {
```

Comment:
this doesn't work properly for `NaN` or `-0` - we should pull in https://www.npmjs.com/package/object-is for this.